libs/base/Control/Monad/Reader/Interface.idr

```diff
@@ -8,14 +8,16 @@ import Control.Monad.RWS.CPS
 import Control.Monad.Trans
 import Control.Monad.Writer.CPS
 
+%default total
+
 ||| A computation which runs in a static context and produces an output
 public export
 interface Monad m => MonadReader stateType m | m where
   ||| Get the context
   ask : m stateType
 
   ||| `local f c` runs the computation `c` in an environment modified by `f`.
-  local : MonadReader stateType m => (stateType -> stateType) -> m a -> m a
+  local : (stateType -> stateType) -> m a -> m a
```

Comment:
I am not familiar with this interface/code however since a constraint is being removed (of its own interface) without issues I will defer to @buzden opinion on that front.

Reply:
Well, not completely wrong. The current `local` function has the signature of
```idris
MonadReader stateType m => MonadReader stateType m => (stateType -> stateType) -> m a -> m a
```
which obviously has no problem at the callsite (of course, unless you put explicitly one of the `auto`-arguments), and I guess at the implementation side no ambiguity was because the current interface it treated specially in functions of interfaces' implementations.